ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/tree/nodes/tables/TomlTablePrimitive.kt

```diff
@@ -17,57 +15,34 @@ import com.akuleshov7.ktoml.writers.TomlEmitter
  * for example: if the TomlTable is [a.b.c] this list will contain [a], [a.b], [a.b.c]
  * @property isSynthetic - flag to determine that this node was synthetically and there is no such table in the input
  */
-@Suppress("MULTIPLE_INIT_BLOCKS")
 public class TomlTablePrimitive(
-    content: String,
+    fullTableKey: TomlKey,
     lineNo: Int,
-    comments: List<String> = emptyList(),
-    inlineComment: String = "",
-    config: TomlInputConfig = TomlInputConfig(),
+    comments: List<String>,
+    inlineComment: String,
     isSynthetic: Boolean = false
 ) : TomlTable(
-    content,
+    fullTableKey,
     lineNo,
     comments,
     inlineComment,
-    config,
     isSynthetic
 ) {
     public override val type: TableType = TableType.PRIMITIVE
 
-    // short table name (only the name without parental prefix, like a - it is used in decoder and encoder)
-    override val name: String
-
-    // list of tables (including sub-tables) that are included in this table  (e.g.: {a, a.b, a.b.c} in a.b.c)
-    public override lateinit var tablesList: List<String>
-
-    // full name of the table (like a.b.c.d)
-    public override lateinit var fullTableName: String
-
-    init {
```

Comment:
I see that you moved this part of code to static method in `TomlTable` - is done for some unification or some other ideas under the hood?

Reply:
Yes